domain/src/test/java/org/oppia/android/domain/classify/rules/fractioninput/FractionInputIsEquivalentToAndInSimplestFormRuleClassifierProviderTest.kt

```diff
@@ -0,0 +1,283 @@
+    val answer = FRACTION_VALUE_TEST_1_OVER_2
+
+    val matches =
+      inputIsEquivalentToAndInSimplestFormRuleClassifier.matches(answer = answer, inputs = inputs)
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testEquivalentAndSimplest_mixedNum16And1Over2Answer_mixedNum6And1Over2Input_matches() {
+    val inputs = mapOf("f" to MIXED_NUMBER_VALUE_TEST_16_AND_1_OVER_2)
+    val answer = MIXED_NUMBER_VALUE_TEST_16_AND_1_OVER_2
+
+    val matches =
+      inputIsEquivalentToAndInSimplestFormRuleClassifier.matches(answer = answer, inputs = inputs)
+
+    assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testEquivalentAndSimplest_mixedNum16And1Over2Answer_mixedNum7And1Over7Input_doesNotMatch() {
+    val inputs = mapOf("f" to MIXED_NUMBER_VALUE_TEST_7_AND_1_OVER_7)
+    val answer = MIXED_NUMBER_VALUE_TEST_16_AND_1_OVER_2
+
+    val matches =
+      inputIsEquivalentToAndInSimplestFormRuleClassifier.matches(answer = answer, inputs = inputs)
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testEquivalentAndSimplest_mixedNum16And2Over4Answer_mixedNum6And1Over2Input_doesNotMatch() {
+    val inputs = mapOf("f" to MIXED_NUMBER_VALUE_TEST_16_AND_1_OVER_2)
+    val answer = MIXED_NUMBER_VALUE_TEST_16_AND_2_OVER_4
+
+    val matches =
+      inputIsEquivalentToAndInSimplestFormRuleClassifier.matches(answer = answer, inputs = inputs)
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testEquivalentAndSimplest_mixedNum16And1Over2Ans_negMixedNum16And1Over2Input_doesNotMatch() {
+    val inputs = mapOf("f" to NEGATIVE_MIXED_NUMBER_VALUE_TEST_16_AND_1_OVER_2)
+    val answer = MIXED_NUMBER_VALUE_TEST_16_AND_1_OVER_2
+
+    val matches =
+      inputIsEquivalentToAndInSimplestFormRuleClassifier.matches(answer = answer, inputs = inputs)
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testEquivalentAndSimplest_missingInputF_throwsException() {
+    val inputs = mapOf("y" to FRACTION_VALUE_TEST_1_OVER_2)
+
+    val exception = assertThrows(IllegalStateException::class) {
+      inputIsEquivalentToAndInSimplestFormRuleClassifier.matches(
+        answer = FRACTION_VALUE_TEST_1_OVER_2,
+        inputs = inputs
+      )
+    }
+
+    assertThat(exception)
+      .hasMessageThat()
+      .contains("Expected classifier inputs to contain parameter with name 'f' but had: [y]")
+  }
+
+  private fun setUpTestApplicationComponent() {
+    DaggerFractionInputIsEquivalentToAndInSimplestFormRuleClassifierProviderTest_TestApplicationComponent // ktlint-disable max-line-length
+      .builder()
+      .setApplication(ApplicationProvider.getApplicationContext()).build().inject(this)
+  }
+
+  private fun <T : Throwable> assertThrows(type: KClass<T>, operation: () -> Unit): T {
+    try {
+      operation()
+      fail("Expected to encounter exception of $type")
+    } catch (t: Throwable) {
+      if (type.isInstance(t)) {
+        return type.cast(t)
+      }
+      // Unexpected exception; throw it.
+      throw t
+    }
+  }
```

Comment:
If we are seeing this being used often, should we keep this function in the testing module? @BenHenning @anandwana001 thoughts?

Reply:
Yes, I guess, I will create a good first issue for this.

Reply:
Probably. :) Thanks @anandwana001. Can you reference that starter issue here once created?

Reply:
https://github.com/oppia/oppia-android/issues/2234